### PR TITLE
[FEA] Add `nlargest`, `nsmallest` series binding

### PR DESCRIPTION
**Pull request by mzegar**
https://github.com/rapidsai/cudf/blob/90fe10b2947c9caee998f6236d432f3742c3504e/python/cudf/cudf/core/series.py#L3242

### CODE REVIEW

File: modules/cudf/src/series.ts [hunk 1]
```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
     return this.__construct(this._col.gather(selection._col));
   }
 
+  /**
+   * perform nLargest or nSmallest operation
+   * @ignore
+   */
+  _nLargestOrSmallest(ascending: boolean, n: number, keep: DuplicateKeepOption) {
```
Comment (trxcllnt) [inline modules/cudf/src/series.ts:655]:
```suggestion
  protected _nLargestOrSmallest(ascending: boolean, n: number, keep: DuplicateKeepOption) {
```

File: modules/cudf/src/series.ts [hunk 2]
```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
...
+    if (keep == DuplicateKeepOption.first) {
+      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
+    } else if (keep == DuplicateKeepOption.last) {
+      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
+                    : this.sortValues(ascending).tail(n).reverse();
+    } else {
+      throw new TypeError('keep must be either "first" or "last"');
+    }
+  }
```
Comment (trxcllnt) [inline modules/cudf/src/series.ts:664]: Why not have these logic branches in each of the functions directly?

**Review by trxcllnt — CHANGES_REQUESTED** [review]
Left a question

File: modules/cudf/src/series.ts [hunk 3]
```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
...
+   * const b = Series.new(["foo", "bar", "test"]);
+   *
+   * a.nLargest(); // [12, 10, 8, 6, 4]
+   * b.nLargest(1); // ["test"]
+   * a.nLargest(-1); // []
+   * ```
+   */
+  nLargest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
+    return this._nLargestOrSmallest(true, n, keep);
+  }
```
Comment (trxcllnt) [inline modules/cudf/src/series.ts:686]:
```suggestion
  nLargest(n = 5, keep: keyof typeof DuplicateKeepOption = 'first') {
    return this._nLargestOrSmallest(true, n, DuplicateKeepOption[keep]);
  }
```

File: modules/cudf/src/series.ts [hunk 4]
```diff
@@ -648,6 +648,65 @@ export class AbstractSeries<T extends DataType = any> {
...
+   * const b = Series.new(["foo", "bar", "test"]);
+   *
+   * a.nSmallest(); // [1, 2, 4, 6, 8]
+   * b.nSmallest(1); // ["bar"]
+   * a.nSmallest(-1); // []
+   * ```
+   */
+  nSmallest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
+    return this._nLargestOrSmallest(false, n, keep);
+  }
```
Comment (trxcllnt) [inline modules/cudf/src/series.ts:708]:
```suggestion
  nSmallest(n = 5, keep: keyof typeof DuplicateKeepOption = 'first') {
    return this._nLargestOrSmallest(false, n, DuplicateKeepOption[keep]);
  }
```

File: modules/cudf/src/series.ts (hunk 2 above)
Comment (mzegar) [reply modules/cudf/src/series.ts:664]: Do you mean...

```ts
nLargest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
    if (keep == DuplicateKeepOption.first) {
      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
    } else if (keep == DuplicateKeepOption.last) {
      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
                    : this.sortValues(ascending).tail(n).reverse();
    } else {
      throw new TypeError('keep must be either "first" or "last"');
    }
}

nSmallest(n = 5, keep: DuplicateKeepOption = DuplicateKeepOption.first) {
    if (keep == DuplicateKeepOption.first) {
      return this.sortValues(!ascending).head(n < 0 ? 0 : n);
    } else if (keep == DuplicateKeepOption.last) {
      return n <= 0 ? Series.new({type: this.type, data: new Array(0)})
                    : this.sortValues(ascending).tail(n).reverse();
    } else {
      throw new TypeError('keep must be either "first" or "last"');
    }
}

```
(ignoring any syntax errors / incorrect variables).

I basically set it up as it is in `CUDF` https://github.com/rapidsai/cudf/blob/90fe10b2947c9caee998f6236d432f3742c3504e/python/cudf/cudf/core/series.py#L3242

I don't really have a preference, I'm fine to change it.

File: modules/cudf/src/series.ts (hunk 3 above)
Comment (mzegar) [reply modules/cudf/src/series.ts:686]: What's the reasoning behind preferring this over the enum? Keeping it similar to how it's used in python?

Comment (trxcllnt) [reply modules/cudf/src/series.ts]: Yeah that, and to match how we're doing it in other places

File: modules/cudf/src/series.ts (hunk 2 above)
Comment (trxcllnt) [reply modules/cudf/src/series.ts:664]: Oh my bad, I didn't realize the switching on `keep` wasn't about the largest/smallest, but the first/last.

File: modules/cudf/src/series.ts (hunk 1 above)
Comment (trxcllnt) [reply modules/cudf/src/series.ts:655]: We could also just make this a utility function at the bottom of the file. Then it's not on the `Series` prototype when people look at series instances in the debugger.

**Review by trxcllnt — COMMENTED** [review]
I was wrong, I think you had it right the first time :sweat_smile:

**Comment by mzegar** [comment]
> I was wrong, I think you had it right the first time :sweat_smile: 

Np, I'll go with the utility function at the bottom of the file.

**Review by trxcllnt — APPROVED** [review]

**Pull request merged**